src/core/server/saved_objects/migrations/actions/initialize_action.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as TaskEither from 'fp-ts/lib/TaskEither';
+import * as Either from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { ElasticsearchClient } from '../../../elasticsearch';
+import {
+  catchRetryableEsClientErrors,
+  RetryableEsClientError,
+} from './catch_retryable_es_client_errors';
+
+import { FetchIndexResponse, fetchIndices } from './fetch_indices';
+
+const routingAllocationEnable = 'cluster.routing.allocation.enable';
+export interface ClusterRoutingAllocationEnabled {
+  clusterRoutingAllocationEnabled: boolean;
+}
+
+export interface InitActionParams {
+  client: ElasticsearchClient;
+  indices: string[];
+}
+
+export interface ClusterAllocationDisabledError {
+  type: 'cluster_routing_allocation_disabled';
+}
+
+export const checkClusterRoutingAllocationEnabledTask =
+  ({
+    client,
+  }: {
+    client: ElasticsearchClient;
+  }): TaskEither.TaskEither<RetryableEsClientError | ClusterAllocationDisabledError, {}> =>
+  () => {
+    return client.cluster
+      .getSettings({
+        include_defaults: true,
+        flat_settings: true,
+      })
+      .then((settings) => {
+        const clusterRoutingAllocations: string[] =
+          settings?.transient?.[routingAllocationEnable] ??
+          settings?.persistent?.[routingAllocationEnable] ??
+          settings?.defaults?.[routingAllocationEnable] ??
+          [];
+
+        const clusterRoutingAllocationEnabled =
+          clusterRoutingAllocations.length === 0 || !clusterRoutingAllocations.includes('none');
```

Comment:
Good call, this is going to take some time to test manually. I'll update the tests to include those options.
Should we also include multi-node testing in the code?

Reply:
> we'll exit migrations early if routing allocation is anything other than `all`. Is that what you were thinking?

I think so. If we can confirm that any other setting would cause the migration to always fail for a multi-node cluster.

There is a chance that we maybe fail for these settings on single node clusters where it wouldn't have mattered. We can ask elasticsearch and support if there's known use cases for having these settings or if it's just something customers set when they're recovering a cluster.

Reply:
I checked with support and they're not aware of any issues in migrations when using `primaries` or `new_primaries` rather than `all`. I'm torn between being overly permissive (allowing anything other than `none`) and being too restrictive (only allowing `all`).
Being too permissive might mean we end up in a failed state later on when waiting for index yellow status, while being too restrictive might have its own issues.